compiler/rustc_parse/src/parser/item.rs

```diff
@@ -1539,7 +1539,7 @@ impl<'a> Parser<'a> {
                     }
                 }
 
-                if self.token.is_ident() {
+                if self.token.is_ident() || self.token.kind == TokenKind::Pound {
```

Comment:
Should we look ahead for more than 1 token, i.e. check for both `# [` tokens? 

Also, please leave a comment explaining why we're checking for `TokenKind::Pound`

Reply:
Fixed!